src/core/servers-setup.c

```diff
@@ -474,10 +474,15 @@ static int compare_server_setup (CONFIG_NODE *node, SERVER_SETUP_REC *server)
 	chatnet = config_node_get_str(node, "chatnet", NULL);
 	port = config_node_get_int(node, "port", 0);
 
+	if (address == NULL || chatnet == NULL) {
+		return 0;
+	}
+
 	if (g_strcmp0(address, server->address) != 0 ||
 	    g_strcmp0(chatnet, server->chatnet) != 0 ||
```

Comment:
Did you forget these, or...?

Reply:
That's what you get when you manually select the hunks you need...